### ISSUE TITLE
Modified version number results in constant connect + disconnect cycle

### ISSUE THREAD

**Issue by LordofSandvich** [Wulf2k/DaS-PC-MPChan#134]
It works fine if the version numbers match but it does not intelligently avoid matching with players whose version number does not match, resulting in occasional lagspikes from the constant connect and disconnect loop. Most noticeable in places already performance-intensive such as Blighttown.

**Comment by Chronial** [comment]
So you modified the version number of your dark souls client?

**Comment by LordofSandvich** [comment]
Someone did it for me, I don't know any code. From what I remember you have to rewrite three locations in the Dark Souls .exe which is best done through a DLL.

**Comment by Chronial** [comment]
Why would you do that? If you did that, your darksouls is not supported by DSCM. Please do not join DSCM-Net if you changed your version.

**Comment by LordofSandvich** [comment]
There are two overhauls which are using modified version numbers in order to function within online play but separate from the main vanilla playerbase. This way players are not invaded by opponents with modified stats and equipment, or worse, have access to invisible spells.

Consider this issue a request.

**Comment by Wulf2k** [comment]
There's no reason to use DSCM-Net with a modified Dark Souls.  Just use the Connect by Steam ID feature.

If you get a large enough playerbase the DSCM-Net code is available to launch your own server.

It may be worth making it easier to change the server address in DSCM at some point.

**Comment by Chronial** [comment]
We should probably detect changed version numbers and force-disable DSCM-Net.

@LordofSandvich DSCM should never go into a "constant connect and disconnect loop". I don't think the performance issues you are having are caused by DSCM.

**Comment by metal-crow** [comment]
I agree that having a separate server for each overhaul is a good idea, but is a separate client also necessary? I'd be open to making a PR that has DSCM support multiple servers, and it auto-chooses which one to use based on dark souls version number.

**Comment by LordofSandvich** [comment]
Could you DM me what I need to know to set up a server, Wulf? Doesn't need to be massively technical but I'd like to know what I or someone else needs to do for it.
